### DISCUSSION TITLE
How to get the dataset of a remote openthread border router ?

### DISCUSSION THREAD

**Discussion by caipiblack** [openthread/openthread#9754] [Q&A] [answered]
Hello,

It seems that the Apple application (using Matter SDK) are able to commission matter devices on Google Nest thread network.

So for me it means that the Matter SDK from Apple is able to get the dataset of the Google Nest device.

I think the Matter SDK of Apple is firstly doing a mDNS research to find the border routers, but as I know there is not enough informations in advertised packets to allow a Thread connexion.

Questions:

- How is it possible ?
- Is it possible with COAP or something else to get the dataset of a thread network (including network key..) from the LAN ?
- Or is it simply using cloud to cloud API to retrieve the datasets of the border-routers associated with the Google account ?

**Comment by wgtdkp** [comment, marked as answer]
This is currently done by letting Google Home App storing the dataset of the Google Nest Border Router to iOS via the iOS ThreadNetwork APIs: https://developer.apple.com/documentation/threadnetwork/thclient/storecredentials(forborderagent:activeoperationaldataset:completion:).

You can do similar on Android with the Thread GMS API: https://developers.home.google.com/thread

**Comment by caipiblack** [comment]
Thank you so much!!
